### PR TITLE
Deprecating `cpp_default` and `wasm_default` cargo features

### PR DESCRIPTION
**Pull request by robertbastian**
With the disappearance of testdata, these become the same, `["logging"]`. Instead I made `logging` a default feature.

I also moved the WASM logging setup from `wasm_glue` into `ICU4XLogger`, to reduce wasm glue.

### CODE REVIEW

**Review by Manishearth — DISMISSED** [review]

File: ffi/diplomat/src/logging.rs [hunk 1]
```diff
@@ -11,14 +11,44 @@ pub mod ffi {
...
-        /// stdout. Returns `false` if there was already a logger set, or if logging has not been
+        /// Initialize the logger using `simple_logger`, or console.log/warn in WASM.
+        ///
+        /// Returns `false` if there was already a logger set, or if logging has not been
         /// compiled into the platform
         pub fn init_simple_logger() -> bool {
-            #[cfg(not(all(not(target_arch = "wasm32"), feature = "simple_logger")))]
-            unimplemented!("ICU4X binary not compiled with simple_logger support");
+            #[cfg(all(not(target_arch = "wasm32"), not(feature = "simple_logger")))]
+            return false;
```
Comment (sffc) [inline ffi/diplomat/src/logging.rs:20]: We used to fail loudly if this function was called in an environment without logging support. Now we fail quietly. Problem?

Comment (robertbastian) [reply ffi/diplomat/src/logging.rs:20]: Well the docs say we don't fail loudly

**Review by Manishearth — APPROVED** [review]

File: ffi/diplomat/src/logging.rs (hunk 1 above)
Comment (robertbastian) [reply ffi/diplomat/src/logging.rs:20]: Probably best to make this a linker error and update the docs

File: ffi/diplomat/Cargo.toml [hunk 2]
```diff
@@ -147,14 +144,11 @@ serde = { version = "1.0", default-features = false, optional = true }
...
+log = { version = "0.4", optional = true }
 
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
-# Logging is automagical in wasm, we only need this for native
+# Logging uses diplomat_runtime bindings in wasm, we only need this for native
 simple_logger = { version = "4.1.0", optional = true }
-log = { version = "0.4", optional = true }
-
-[target.'cfg(target_arch = "wasm32")'.dependencies]
-log = "0.4"
```
Comment (sffc) [inline ffi/diplomat/Cargo.toml:157]: Issue: Is it correct to say that with this change, logging gets disabled in wasm mode by default? That would be a regression.

File: ffi/gn/icu4x/BUILD.gn [hunk 3]
```diff
@@ -254,9 +235,6 @@ rust_library("icu_capi-v1_2_2") {
   deps += [ ":tinystr-v0_7_1" ]
   deps += [ ":unicode-bidi-v0_3_13" ]
   deps += [ ":writeable-v0_5_2" ]
-  if (current_cpu == "wasm32") {
-    deps += [ ":log-v0_4_17" ]
-  }
```
Comment (sffc) [inline ffi/gn/icu4x/BUILD.gn:259]: Observation: Was logging ever enabled by default in wasm? The `log` crate should be a dependency of `icu_provider`, not `diplomat`

**Review by sffc — CHANGES_REQUESTED** [review]

File: ffi/diplomat/Cargo.toml (hunk 2 above)
Comment (robertbastian) [reply ffi/diplomat/Cargo.toml:157]: No, but gn uses `default-features = false`

File: ffi/gn/icu4x/BUILD.gn (hunk 3 above)
Comment (robertbastian) [reply ffi/gn/icu4x/BUILD.gn:259]: And `icu_capi` itself needs the `log` crate to register the wasm console logger. I'd like to move that into `diplomat_runtime` at some point, but for now it's here.

Comment (robertbastian) [reply ffi/gn/icu4x/BUILD.gn:259]: And `icu_capi` itself needs the `log` crate to register the wasm console logger. I'd like to move that into `diplomat_runtime` at some point, but for now it's here.

**Review by sffc — APPROVED** [review]

**Review by Manishearth — APPROVED** [review]

**Pull request merged**